openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/CollectionDAO.java

```diff
@@ -5516,4 +5526,42 @@ default String getNameHashColumn() {
       return "fqnHash";
     }
   }
+
+  interface WorkflowDefinitionDAO extends EntityDAO<WorkflowDefinition> {
+    @Override
+    default String getTableName() {
+      return "workflow_definition_entity";
+    }
+
+    @Override
+    default Class<WorkflowDefinition> getEntityClass() {
+      return WorkflowDefinition.class;
+    }
+
+    @Override
+    default String getNameHashColumn() {
+      return "fqnHash";
+    }
+  }
+
+  interface WorkflowInstanceTimeSeriesDAO extends EntityTimeSeriesDAO {
+    @Override
+    default String getTimeSeriesTableName() {
+      return "workflow_instance_time_series";
+    }
+  }
+
+  interface WorkflowInstanceStateTimeSeriesDAO extends EntityTimeSeriesDAO {
+    @Override
+    default String getTimeSeriesTableName() {
+      return "workflow_instance_state_time_series";
+    }
+
+    @SqlQuery(
+        value =
+            "SELECT json FROM workflow_instance_state_time_series "
+                + "WHERE entityFQNHash = :entityFQNHash ORDER BY timestamp DESC")
+    List<String> listWorkflowInstanceStatesForWorkflowInstanceId(
+        @Bind("entityFQNHash") String entityFQNHash);
+  }
```

Comment:
Do you think we can use `listWithOffset` from `EntityTimeSeriesDAO` passing the entittyFQNHash as the searchFilter? This way we handle pagination as well.

Reply:
I'll take a look!

Reply:
Done